packages/taxdat/R/postprocessing.R

```diff
@@ -0,0 +1,712 @@
+#'
+#' @return
+#' @export
+#'
+#' @examples
+get_space_grid <- function(config_list,
+                           redo = FALSE) {
+  
+  # Make file name
+  space_grid_sf_file <- stringr::str_replace(config_list$file_names$observations_filename,
+                                             ".preprocess.rdata",
+                                             "_space_grid_sf.rds")
+  
+  if (file.exists(space_grid_sf_file) & !redo) {
+    res <- readRDS(space_grid_sf_file)
+  } else {
+    res <- taxdat::read_file_of_type(config_list$file_names$stan_input_filename, "stan_input")$sf_grid %>% 
+      dplyr::group_by(rid, x, y) %>% 
+      dplyr::slice(1) %>% 
+      dplyr::select(rid, x, y, geom)
+    
+    saveRDS(res, file = space_grid_sf_file)
+  }
+  
+  res
+}
+
+#' get_smooth_grid
+#'
+#' @param config_list 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+get_spacetime_grid <- function(config_list,
+                               redo = FALSE) {
+  
+  # Make file name
+  spacetime_grid_sf_file <- stringr::str_replace(config_list$file_names$observations_filename,
+                                                 ".preprocess.rdata",
+                                                 "_spacetime_grid_sf.rds")
+  
+  if (file.exists(spacetime_grid_sf_file) & !redo) {
+    res <- readRDS(spacetime_grid_sf_file)
+  } else {
+    res <- taxdat::read_file_of_type(config_list$file_names$stan_input_filename, "stan_input")$sf_grid
+    
+    saveRDS(res, file = spacetime_grid_sf_file)
+  }
+  
+  res
+}
+
+get_mean_pop_grid <- function(config_list,
+                              redo = FALSE) {
+  
+  # Make file name
+  mean_pop_grid_sf_file <- stringr::str_replace(config_list$file_names$observations_filename,
+                                                ".preprocess.rdata",
+                                                "_mean_pop_grid_sf.rds")
+  
+  if (file.exists(mean_pop_grid_sf_file) & !redo) {
+    res <- readRDS(mean_pop_grid_sf_file)
+  } else {
+    res <- get_spacetime_grid(config_list = config_list) %>% 
+      mutate(pop = taxdat::read_file_of_type(config_list$file_names$stan_input_filename, variable = "stan_input")$stan_data$pop) %>% 
+      sf::st_drop_geometry() %>% 
+      dplyr::group_by(rid, x, y, id) %>% 
+      dplyr::summarise(pop = mean(pop)) %>% 
+      dplyr::inner_join(get_space_grid(config_list = config_list), .)
+    
+    saveRDS(res, file = mean_pop_grid_sf_file)
+  }
+  
+  res
+}
+
+
+# Postprocess functions ---------------------------------------------------
+
+#' collapse_grid
+#' Function to collapse space grid to single non-overlapping cells
+#' 
+#' @param grid_obj 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+collapse_grid <- function(df) {
+  
+  u_grid <- df  %>%
+    group_by(rid, x, y) %>% 
+    slice(1) %>% 
+    select(rid, x, y, geom)
+  
+  
+  res <- df %>%
+    st_drop_geometry() %>% 
+    group_by(rid, x, y) %>% 
+    summarise(mean = mean(mean),
+              q5 = min(q5),
```

Comment:
This reminds me ... Don't we actually want to change the quantiles to 0.25 and 0.975 so that these will represent a 95% Prediction Interval?

Reply:
true, the thing is that the default of summarise_draws is to return th 5 and 95 interval. We would need to change this throughout. I will try addressing this before the merge.

Reply:
this is fixed.